client/a8c-for-agencies/components/a4a-feedback/churn-mechanism/cancel-license-feedback-modal.tsx

```diff
@@ -0,0 +1,333 @@
+
+	return (
+		<A4AModal
+			title={ translate( 'Are you sure you want to revoke this license?' ) }
+			subtile={ translate(
+				'A revoked license cannot be reused, and the associated site will no longer have access to the provisioned product. You will stop being billed for this license immediately.'
+			) }
+			onClose={ onClose }
+			showCloseButton={ false }
+			extraActions={
+				<>
+					<Button
+						isDestructive
+						onClick={ () => handleCTAClick( 'cancel' ) }
+						variant="secondary"
+						disabled={ isLoading }
+						isBusy={ selectedCTA === 'cancel' && isLoading }
+					>
+						{ translate( 'Cancel license' ) }
+						{ isAtomicSite && <Icon icon={ external } size={ 18 } /> }
+					</Button>
+					<Button
+						onClick={ () => handleCTAClick( 'speak-with-partner-manager' ) }
+						variant="primary"
+						disabled={ isLoading }
+						isBusy={ selectedCTA === 'speak-with-partner-manager' && isLoading }
+					>
+						{ translate( 'Speak with my Partner Manager' ) }
+					</Button>
+				</>
+			}
+		>
+			<div className="a4a-feedback__churn-mechanism">
+				<div className="a4a-feedback__license-details">
+					{ siteUrl && (
+						<div className="a4a-feedback__license-details-item">
+							<div className="a4a-feedback__license-details-item-label">
+								{ translate( 'Site:' ) }
+							</div>
+							<div className="a4a-feedback__license-details-item-label">
+								{ isPressableLicense ? translate( 'Multiple sites use this license' ) : siteSlug }
+							</div>
+						</div>
+					) }
+					<div className="a4a-feedback__license-details-item">
+						<div className="a4a-feedback__license-details-item-label">
+							{ translate( 'Product:' ) }
+						</div>
+						<div className="a4a-feedback__license-details-item-value">
+							{ productName }
+							{ bundleSize && (
+								<>
+									{ translate( ' (%(bundleSize)d licenses)', {
+										args: { bundleSize },
+									} ) }
+								</>
+							) }
+						</div>
+					</div>
+					<div className="a4a-feedback__license-details-item">
+						<div className="a4a-feedback__license-details-item-label">
+							{ translate( 'License:' ) }
+						</div>
+						<div className="a4a-feedback__license-details-item-value">{ licenseKey }</div>
+					</div>
+					<div className="a4a-feedback__license-details-warning">
+						<Icon icon={ info } size={ 24 } />
+						{ translate( 'Please note this action cannot be undone.' ) }
+					</div>
+				</div>
+
+				{ suggestion && ! isHostingLicense && (
+					<FormFieldset>
+						<FormLabel className="a4a-feedback__comments-label" htmlFor="suggestion">
+							{ suggestion.label }
+						</FormLabel>
+						<div className="a4a-feedback__suggestions">
+							{ suggestion.options.map( ( option ) => (
+								<CheckboxControl
+									key={ `suggestion-${ option.value }` }
+									label={ option.label }
+									checked={
+										!! suggestions.find( ( suggestion ) => suggestion.value === option.value )
+									}
```

Comment:
```suggestion
									checked={ suggestions.some(
										( suggestion ) => suggestion.value === option.value
									) }
```

We can utilize `array.some` here so we don't need to convert the return type to boolean. It's also more natural to read.

Reply:
Sure! Updated: https://github.com/Automattic/wp-calypso/pull/102496/commits/8d9d435a450b3bc05da8a2b69bf15a70b2f6311f